hudi-aws/src/main/java/org/apache/hudi/aws/sync/AWSGlueCatalogClient.java

```diff
@@ -0,0 +1,483 @@
+      MessageType storageSchema,
+      String inputFormatClass,
+      String outputFormatClass,
+      String serdeClass,
+      Map<String, String> serdeProperties,
+      Map<String, String> tableProperties) {
+    if (tableExists(tableName)) {
+      return;
+    }
+    CreateTableRequest request = new CreateTableRequest();
+    Map<String, String> params = new HashMap<>();
+    if (!syncConfig.createManagedTable) {
+      params.put("EXTERNAL", "TRUE");
+    }
+    params.putAll(tableProperties);
+
+    try {
+      Map<String, String> mapSchema = parquetSchemaToMapSchema(storageSchema, syncConfig.supportTimestamp, false);
+
+      List<Column> schemaPartitionKeys = new ArrayList<>();
+      List<Column> schemaWithoutPartitionKeys = new ArrayList<>();
+      for (String key : mapSchema.keySet()) {
+        String keyType = getPartitionKeyType(mapSchema, key);
+        Column column = new Column().withName(key).withType(keyType.toLowerCase()).withComment("");
+        // In Glue, the full schema should exclude the partition keys
+        if (syncConfig.partitionFields.contains(key)) {
+          schemaPartitionKeys.add(column);
+        } else {
+          schemaWithoutPartitionKeys.add(column);
+        }
+      }
+
+      StorageDescriptor storageDescriptor = new StorageDescriptor();
+      serdeProperties.put("serialization.format", "1");
+      storageDescriptor
+          .withSerdeInfo(new SerDeInfo().withSerializationLibrary(serdeClass).withParameters(serdeProperties))
+          .withLocation(s3aToS3(syncConfig.basePath))
+          .withInputFormat(inputFormatClass)
+          .withOutputFormat(outputFormatClass)
+          .withColumns(schemaWithoutPartitionKeys);
+
+      TableInput tableInput = new TableInput()
+          .withName(tableName)
+          .withTableType(TableType.EXTERNAL_TABLE.toString())
+          .withParameters(params)
+          .withPartitionKeys(schemaPartitionKeys)
+          .withStorageDescriptor(storageDescriptor)
+          .withLastAccessTime(new Date(System.currentTimeMillis()))
+          .withLastAnalyzedTime(new Date(System.currentTimeMillis()));
+      request.withDatabaseName(databaseName)
+          .withTableInput(tableInput);
+
+      CreateTableResult result = awsGlue.createTable(request);
+      LOG.info("Created table " + tableId(databaseName, tableName) + " : " + result);
+    } catch (AlreadyExistsException e) {
+      LOG.warn("Table " + tableId(databaseName, tableName) + " already exists.", e);
+    } catch (Exception e) {
+      throw new HoodieGlueSyncException("Fail to create " + tableId(databaseName, tableName), e);
+    }
+  }
+
+  @Override
+  public Map<String, String> getTableSchema(String tableName) {
+    try {
+      // GlueMetastoreClient returns partition keys separate from Columns, hence get both and merge to
+      // get the Schema of the table.
+      final long start = System.currentTimeMillis();
+      Table table = getTable(awsGlue, databaseName, tableName);
```

Comment:
In HoodieHiveClient first we check tableExists() and then proceed further. Do you want to follow the same pattern here. will let you take the call.